crates/backend/src/codegen.rs

```diff
@@ -146,131 +146,180 @@ impl ToTokens for ast::Struct {
         let name_str = self.js_name.to_string();
         let name_len = name_str.len() as u32;
         let name_chars = name_str.chars().map(|c| c as u32);
-        let new_fn = Ident::new(&shared::new_function(&name_str), Span::call_site());
-        let free_fn = Ident::new(&shared::free_function(&name_str), Span::call_site());
-        let free_fn_const = Ident::new(&format!("{}__const", free_fn), free_fn.span());
-        (quote! {
-            #[automatically_derived]
-            impl wasm_bindgen::describe::WasmDescribe for #name {
-                fn describe() {
-                    use wasm_bindgen::__wbindgen_if_not_std;
-                    __wbindgen_if_not_std! {
-                        compile_error! {
-                            "exporting a class to JS requires the `std` feature to \
-                             be enabled in the `wasm-bindgen` crate"
-                        }
+        if self.transparent {
+            let field = &self.fields[0];
+            let field_ty = &field.ty;
+            (quote! {
+                #[automatically_derived]
+                impl wasm_bindgen::describe::WasmDescribe for #name {
+                    fn describe() {
+                        #field_ty::describe();
                     }
-                    use wasm_bindgen::describe::*;
-                    inform(RUST_STRUCT);
-                    inform(#name_len);
-                    #(inform(#name_chars);)*
                 }
-            }
 
-            #[automatically_derived]
-            impl wasm_bindgen::convert::IntoWasmAbi for #name {
-                type Abi = u32;
+                #[automatically_derived]
+                impl wasm_bindgen::convert::IntoWasmAbi for #name {
+                    type Abi = u32;
```

Comment:
I don't think that `u32` is right here, instead I think it should be `<#field_ty as IntoWasmAbi>::Abi` (and similar for other occurrences below)